blueprints/data-solutions/data-platform-minimal/variables.tf

```diff
@@ -0,0 +1,225 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# tfdoc:file:description Terraform Variables.
+
+variable "composer_config" {
+  description = "Cloud Composer config."
+  type = object({
+    disable_deployment = optional(bool)
+    environment_size   = optional(string, "ENVIRONMENT_SIZE_SMALL")
+    software_config = optional(object({
+      airflow_config_overrides = optional(any)
+      pypi_packages            = optional(any)
+      env_variables            = optional(map(string))
+      image_version            = string
+      }), {
+      image_version = "composer-2-airflow-2"
+    })
+    workloads_config = optional(object({
+      scheduler = optional(object(
+        {
+          cpu        = number
+          memory_gb  = number
+          storage_gb = number
+          count      = number
+        }
+        ), {
+        cpu        = 0.5
+        memory_gb  = 1.875
+        storage_gb = 1
+        count      = 1
+      })
+      web_server = optional(object(
+        {
+          cpu        = number
+          memory_gb  = number
+          storage_gb = number
+        }
+        ), {
+        cpu        = 0.5
+        memory_gb  = 1.875
+        storage_gb = 1
+      })
+      worker = optional(object(
+        {
+          cpu        = number
+          memory_gb  = number
+          storage_gb = number
+          min_count  = number
+          max_count  = number
+        }
+        ), {
+        cpu        = 0.5
+        memory_gb  = 1.875
+        storage_gb = 1
+        min_count  = 1
+        max_count  = 3
+      })
+    }))
+  })
+  default = {
```

Comment:
- Remove `map(any)`
- You might also want `nullable = false`
- why nest `env_variables` (and siblings)?
- I'd avoid the default values and just let the provider set them.

In summary, I'd use something like this:
```hcl
variable "composer_config" {
  description = "Cloud Composer config."
  type = object({
    disable_deployment = optional(bool, false)
    environment_size   = optional(string, "ENVIRONMENT_SIZE_SMALL")
    software_config = optional(object({
      airflow_config_overrides = optional(map(string), {})
      pypi_packages            = optional(map(string), {})
      env_variables            = optional(map(string), {})
      image_version            = optional(string, "composer-2-airflow-2")
    }))
    scheduler_config = optional(object({
      cpu        = optional(number, 0.75)
      memory_gb  = optional(number, 1.875)
      storage_gb = optional(number, 1)
      count      = optional(number, 1)
    }), {})
    web_server_config = optional(object({
      cpu               = optional(number, 0.5)
      memory_gb         = optional(number, 1.875)
      storage_gb        = optional(number, 1)
      allowed_ip_ranges = map(string)
    }), {})
    worker_config = optional(object({
      cpu        = optional(number, 0.5)
      memory_gb  = optional(number, 1.875)
      storage_gb = optional(number, 1)
      min_count  = optional(number, 1)
      max_count  = optional(number, 3)
    }), {})
  })
  nullable = false
  default  = {}
}


```